src/partition/ReceivedPartition.hpp

```diff
@@ -95,6 +98,9 @@ class ReceivedPartition : public Partition {
 
   /// Min global vertex IDs of remote connected ranks
   std::vector<int> _remoteMinGlobalVertexIDs;
+
+  /// Make the fixture friend of this class
+  friend struct receivedPartitionFixture;
```

Comment:
I'm just using the same pattern in #1050. I think the `friend` declaration should go before the access modifiers. Then you also directly see the friends. This looks more reasonable to me, since a private or public friend is a bit odd (even though I believe it defaults to private?). I would suggest to follow this structure:
```
class A {
friend struct friendOfA;
public:
    ...
private:
    ...
};
```

I found a similar code snipped here: https://docs.microsoft.com/en-us/cpp/cpp/friend-cpp?view=msvc-160.

However, there is also the structure you are using shown in cppreference. See https://en.cppreference.com/w/cpp/language/friend. In the end it seems to be a matter of taste. Any opinions?

Reply:
Yes, it seems more natural to me as well. I changed the declaration location as you suggest. 
One more thing: I separate the declaration and implementation of the fixture. I use a separate hpp file for that, which I think helps readability.